example/src/new_api/velocityTest/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { StyleSheet, View } from 'react-native';
+import Animated, {
+  measure,
+  useAnimatedRef,
+  useAnimatedStyle,
+  useSharedValue,
+  withDecay,
+} from 'react-native-reanimated';
+
+import React from 'react';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+
+const BOX_SIZE = 120;
+
+export default function App() {
+  const aref = useAnimatedRef() as React.RefObject<View>;
+  const offsetX = useSharedValue(0);
+  const offsetY = useSharedValue(0);
+
+  const pan = Gesture.Pan()
+    .onChange((event) => {
+      offsetX.value += event.changeX;
+      offsetY.value += event.changeY;
+    })
+    .onFinalize((event) => {
+      const size = measure(aref)!;
+
+      offsetX.value = withDecay({
+        velocity: event.velocityX,
+        clamp: [-size.width / 2 + BOX_SIZE / 2, size.width / 2 - BOX_SIZE / 2],
+      });
+
+      offsetY.value = withDecay({
+        velocity: event.velocityY,
+        clamp: [
+          -size.height / 2 + BOX_SIZE / 2,
+          size.height / 2 - BOX_SIZE / 2,
+        ],
+      });
+    });
+
+  const animatedStyles = useAnimatedStyle(() => ({
+    transform: [{ translateX: offsetX.value }, { translateY: offsetY.value }],
+  }));
+
+  return (
+    <View style={styles.container} ref={aref}>
```

Comment:
wow, I didn't know that you can use animated refs on non-animated views, thanks! 🤯

Reply:
Ah yes, the totally-on-purpose animated ref on non-animated view. Happy to help.